src/Symfony/Component/ExpressionLanguage/ParsedExpression.php

```diff
@@ -42,6 +42,17 @@ public function getNodes()
 
     public function dump()
     {
-        return $this->nodes->dump();
+        return $this->dumpNode($this->nodes);
+    }
+
+    protected function dumpNode(Node $node)
```

Comment:
`protected` on purpose since this is exactly the extension point that is missing currently to hook into dumping each node